### PR TITLE
refactor(subnav)!: convert to CSS modules

### PR DESCRIPTION
**Pull request by zchsh**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1199185162145306/f)
🔍 [Preview Link](https://react-components-git-zssubnav-css-modules-hashicorp.vercel.app/components/subnav)

---

This PR updates `@hashicorp/react-subnav` to use CSS modules.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/subnav/partials/TitleLink/index.js [hunk 1]
```diff
@@ -31,7 +33,7 @@ function TitleLink(props) {
   return (
     <LinkWrap
       Link={Link}
-      className={`title-link brand-${product} ${!Logo ? 'is-text' : ''}`}
+      className={classNames(s.root, s[`brand-${product}`])}
```
Comment (zchsh) [inline packages/subnav/partials/TitleLink/index.js:36]: `brand-product` class has been retained here for a special case related to sizing the HCP logo.

File: pages/global.css [hunk 2]
```diff
@@ -6,7 +6,6 @@
 @import '../packages/callouts/styles/index.css';
 @import '../packages/checkbox-input/style.css';
 @import '../packages/accordion/style.css';
-@import '../packages/subnav/style.css';
```
Comment (zchsh) [inline pages/global.css:9]: 🔪 🎉

File: packages/subnav/partials/CtaLinks/github-stars-link/index.js [hunk 3]
```diff
@@ -41,27 +43,28 @@ function GithubStarsButton(props) {
...
       </span>
-      {showStarcount && (
-        <span className="stars">
+      {showStarCount && (
+        <span className={s.stars}>
           <StarIcon />
           <span className="g-type-body-small-strong" data-testid="github-stars">
             {formatStarCount(starCount) || <span>&mdash;</span>}
           </span>
+          <VisuallyHidden>Stars on GitHub</VisuallyHidden>
```
Comment (BRKalow) [inline packages/subnav/partials/CtaLinks/github-stars-link/index.js:67]: Great enhancement 👏

File: packages/subnav/partials/MenuItemsDefault/style.module.css [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  white-space: nowrap;
+  margin: 0;
+  padding: 0;
+}
+
+.navLink {
+  composes: g-type-body-small-strong from global;
+  position: relative;
+  padding-left: 16px;
+  padding-right: 16px;
```
Comment (BRKalow) [inline packages/subnav/partials/MenuItemsDefault/style.module.css:25]: nit:
```suggestion
  padding: 0 16px;
```

File: packages/subnav/partials/MenuItemsDefault/style.module.css [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.submenuItem {
+  composes: g-type-body-small-strong from global;
+  color: var(--black);
+}
+
+/*
+Note: duplicated in MenuItemsDefault
+*/
```
Comment (BRKalow) [inline packages/subnav/partials/MenuItemsDefault/style.module.css:39]: This seems like a good indication to abstract the pattern into a component, at least internally within `Subnav` 💭

File: packages/subnav/partials/MenuItemsDefault/style.module.css [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    display: none;
+  }
+}
+
+.verticalDivider {
+  background: var(--gray-5);
+  height: 1.75rem;
+  width: 1px;
+  margin-left: 8px;
+  margin-right: 8px;
```
Comment (BRKalow) [inline packages/subnav/partials/MenuItemsDefault/style.module.css:94]: nit:
```suggestion
  margin: 0 8px;
```

File: packages/subnav/style.module.css [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+.subnavInner {
+  display: flex;
+  align-items: center;
+  height: 62px;
+
+  &.notConstrained {
+    padding-left: 24px;
+    padding-right: 24px;
```
Comment (BRKalow) [inline packages/subnav/style.module.css:24]: nit:
```suggestion
    padding: 0 24px;
```

File: packages/subnav/style.module.css [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    @media (--medium-up) {
+      padding-left: 32px;
+      padding-right: 32px;
```
Comment (BRKalow) [inline packages/subnav/style.module.css:28]: nit:
```suggestion
      padding: 0 32px;
```

**Review by BRKalow — APPROVED** [review]
Looks good! Left some very minor nits on some potential CSS property cleanup. Feel free to ignore!

File: packages/subnav/partials/MenuItemsDefault/style.module.css (hunk 5 above)
Comment (zchsh) [reply packages/subnav/partials/MenuItemsDefault/style.module.css]: 💖  Great call - ✅ abstracted out into a mostly-just-styles `NavItemText` component.

File: packages/subnav/style.module.css (hunk 7 above)
Comment (zchsh) [reply packages/subnav/style.module.css]: 💯 Thanks for making these suggestions - tried out GitHub's "batch suggestions" thing for the first time, worked real nice 👨‍🍳

**Pull request merged**